docker/Dockerfile

```
@@ -1,62 +1,95 @@
-# Install system dependencies
-RUN apt-get update && \
+# Create pyhss folder Install system dependencies for python setuptool + wheels
+RUN mkdir -p /opt/pyhss && \
+    apt-get update && \
     apt-get install -y --no-install-recommends \
-        python3 \
-        python3-dev \
-        python3-venv \
-        python3-pip \
-        libcurl4-openssl-dev \
-        libmysqlclient-dev \
+        libmariadb-dev \
         libsctp-dev \
         lksctp-tools \
-        build-essential \
         pkg-config \
-        libffi-dev \
-        libssl-dev \
-        libxml2-dev \
-        libxslt-dev \
-        libyaml-dev \
-        git \
-        default-libmysqlclient-dev \
-        mysql-server \
-        lksctp-tools && \
+        gcc && \
     apt-get clean && \
-    rm -rf /var/lib/apt/lists/* && \
-    rm -f /usr/lib/python3.*/EXTERNALLY-MANAGED
+    rm -rf /var/lib/apt/lists/*
 
+# Copy requirements.txt early to have cache hits
+COPY requirements.txt /opt/pyhss
 # Create and activate virtual environment
-RUN python3 -m venv /opt/venv
-ENV PATH="/opt/venv/bin:$PATH"
+RUN python3 -m venv /opt/pyhss/venv
+ENV PATH="/opt/pyhss/venv/bin:$PATH"
 
-#Create users on MySQL
-RUN service mysql start && \
-    #mysql -e "CREATE DATABASE IF NOT EXISTS hss;" && \
-    mysql -e "CREATE USER 'dbeaver'@'localhost' IDENTIFIED BY 'password';" && \
-    mysql -e "GRANT ALL PRIVILEGES ON hss.* TO 'dbeaver'@'localhost';" && \
-    mysql -e "FLUSH PRIVILEGES;" 
+# Install Python dependencies
+RUN pip install --upgrade pip setuptools wheel
+RUN pip install --no-cache-dir -r /opt/pyhss/requirements.txt
 
-# Bypass PEP 668 protection (like sudo pip install)
-RUN echo "[global]\nbreak-system-packages = true" > /etc/pip.conf
+FROM python:3.12-slim-trixie AS pyhss-runtime
 
-# Clone repo
-RUN git clone https://github.com/nickvsnetworking/pyhss.git
-#Checkout to the latest commit
-WORKDIR /pyhss
-RUN git checkout DockerFixes
+ENV PYTHONDONTWRITEBYTECODE=1 \
+    PYTHONUNBUFFERED=1 \
+    DEBIAN_FRONTEND=noninteractive
 
-RUN pip install --upgrade pip setuptools wheel
-RUN pip install --no-cache-dir -r /pyhss/requirements.txt
+# Install system dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        libmariadb3 \
+        libsctp1 \
+        lksctp-tools \
+        curl \
+        netcat-traditional && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Copy envsubst from the previous stage
+COPY --from=envsubst /usr/local/bin/envsubst /usr/local/bin/envsubst
 
-WORKDIR /pyhss/services
+# Create application user and group
+RUN mkdir -p /opt/pyhss && \
+    groupadd -r pyhss && \
+    useradd -d /opt/pyhss -r -g pyhss pyhss && \
+    chown -R pyhss:pyhss /opt/pyhss
 
-# Expose ports
+# Copy Python virtual environment from the previous stage and set PATH
+ENV PATH="/opt/pyhss/venv/bin:$PATH"
+COPY --from=python-venv /opt/pyhss/venv /opt/pyhss/venv
+
+# Copy application code
+USER pyhss
+COPY --chown=pyhss:pyhss . /opt/pyhss
```

we don't want pyhss to change any of its own code, so:
```suggestion
COPY . /opt/pyhss
```